projects/plugins/jetpack/_inc/blogging-prompts.php

```diff
@@ -47,6 +47,33 @@ function jetpack_setup_blogging_prompt_response( $post_id ) {
 
 add_action( 'wp_insert_post', 'jetpack_setup_blogging_prompt_response' );
 
+/**
+ * Add a meta value to posts when publishd that answer a blogging prompt.
+ *
+ * @param string  $new_status New post status.
+ * @param string  $old_status Old post status.
+ * @param WP_Post $post       Post object.
+ */
+function jetpack_mark_if_post_answers_blogging_prompt( $new_status, $old_status, $post ) {
+	// Make sure we are publishing a post, and it's not already published.
+	if ( 'post' !== $post->post_type || 'publish' !== $new_status || 'publish' === $old_status ) {
+		return;
+	}
+
+	$blocks = parse_blocks( $post->post_content );
+	foreach ( $blocks as $block ) {
+		if ( 'jetpack/blogging-prompt' === $block['blockName'] ) {
+			$prompt_id = $block['attrs']['promptId'];
+
+			if ( $prompt_id && count( $blocks ) > 1 ) {
+				update_post_meta( $post->ID, '_jetpack_blogging_prompt_key', $prompt_id );
```

Comment:
maybe we should add the tag here as well (what do you think?)
```suggestion
				update_post_meta( $post->ID, '_jetpack_blogging_prompt_key', $prompt_id );
				wp_add_post_tags( $post->ID, array( 'dailyprompt', "dailyprompt-$prompt_id" ) );

```

Reply:
I don't actually want to add the tag server-side, as this will force it upon users. I'll try adding this client side, so it can easily be removed when the post is published.

Reply:
that makes sense, but it opens the question:

If a user has explicitly removed that tag, should we still add the post meta-data.

It seems to me that the same logic should apply to the metadata as applies to the tags right?

Reply:
Without thinking about it too much, that makes sense. I'll plan to add a check that only adds the meta if the tag is present.